sklearn/base.py

```diff
@@ -1476,3 +1477,38 @@ def wrapper(estimator, *args, **kwargs):
         return wrapper
 
     return decorator
+
+
+@dataclass
+class ObjectiveFunction:
+    """A class to represent the objective function of a model.
+
+    Parameters
+    ----------
+    name : str
+        The name of the objective function.
+
+    value : float
+        The total value of the objective function.
+
+    goal : {"minimize", "maximize"}
+        Whether the model tries to minimize or maximize the objective function.
+
+    data_fit : float
+        The term in the objective function that measures the discrepancy between the
+        model's predictions and the true values.
+
+    penalisations : dict
+        The terms in the objective function that measure the penalty for the model.
+
+    normalized : bool
+        Whether each term and the total value of the objective function is
+        per-sample or not.
```

Comment:
Personally I always normalize the objective function, and I would not even add this parameter here.
The reason I added it is that usually in the literature and even in our docstrings and user guides the formula of the objective function is not normalized, so I thought that it might be less confusing for some people. (although they just have to multiply by n_samples).

If we're all on the same page, I'd be in favor of not having the param and always normalize.

Reply:
I suppose that if the docstring is explicit enough, it's fine to always normalize by sample size.